src/script/api/script_date.hpp

```diff
@@ -12,7 +12,148 @@
+	{
+		return new ScriptDateBase(static_cast<Date>(Timer::date.base()));
+	}
+
+	SQInteger GetYear()
+	{
+		if (!this->IsValid()) return DATE_INVALID;
+
+		YearMonthDay ymd = Timer::ConvertDateToYMD(static_cast<Timer::Date>(this->_date));
+		return ymd.year.base();
+	}
+
+	SQInteger GetMonth()
+	{
+		if (!this->IsValid()) return DATE_INVALID;
+
+		YearMonthDay ymd = Timer::ConvertDateToYMD(static_cast<Timer::Date>(this->_date));
+		return ymd.month + 1;
+	}
+
+	SQInteger GetDayOfMonth()
+	{
+		if (!this->IsValid()) return DATE_INVALID;
+
+		YearMonthDay ymd = Timer::ConvertDateToYMD(static_cast<Timer::Date>(this->_date));
+		return ymd.day;
+	}
+
+	static ScriptDateBase *GetDate(SQInteger year, SQInteger month, SQInteger day_of_month)
+	{
+		if (month < 1 || month > 12) return new ScriptDateBase();
+		if (day_of_month < 1 || day_of_month > 31) return new ScriptDateBase();
+
+		Year timer_year{ClampTo<int32_t>(year)};
+		if (timer_year < Time::MIN_YEAR || timer_year > Time::MAX_YEAR) return new ScriptDateBase();
+
+		return new ScriptDateBase(static_cast<Date>(Timer::ConvertYMDToDate(timer_year, month - 1, day_of_month).base()));
+	}
+
+	static SQInteger GetSystemTime()
+	{
+		time_t t;
+		time(&t);
+		return t;
+	}
+
+	Date date() { return this->_date; }
+
+	SQInteger _tostring(HSQUIRRELVM vm)
+	{
+		sq_pushstring(vm, fmt::format("{}", this->_date));
+		return 1;
+	}
+
+	SQInteger _add(HSQUIRRELVM vm)
+	{
+		SQInteger days = 0;
+		switch (sq_gettype(vm, 2)) {
+			case OT_INTEGER:
+				sq_getinteger(vm, 2, &days);
+				break;
+			case OT_INSTANCE: {
+				ScriptDateBase *other = static_cast<ScriptDateBase *>(Squirrel::GetRealInstance(vm, 2, Tag.value));
+				if (other == nullptr) return SQ_ERROR;
+				days = other->_date;
+				break;
+			}
+			default: return SQ_ERROR;
+		}
+		ScriptDateBase *res = new ScriptDateBase(static_cast<Date>(this->_date + days));
+		res->AddRef();
+		Squirrel::CreateClassInstanceVM(vm, Tag.value, res, nullptr, SQConvert::DefSQDestructorCallback<ScriptDateBase>, true);
+		return 1;
+	}
+
+	SQInteger _sub(HSQUIRRELVM vm)
+	{
+		SQInteger days = 0;
+		switch (sq_gettype(vm, 2)) {
+			case OT_INTEGER:
+				sq_getinteger(vm, 2, &days);
+				break;
+			case OT_INSTANCE: {
+					ScriptDateBase *other = static_cast<ScriptDateBase *>(Squirrel::GetRealInstance(vm, 2, Tag.value));
+					if (other == nullptr) return SQ_ERROR;
+					days = other->_date;
+					break;
+			}
+			default: return SQ_ERROR;
+		}
+		ScriptDateBase *res = new ScriptDateBase(static_cast<Date>(this->_date - days));
+		res->AddRef();
+		Squirrel::CreateClassInstanceVM(vm, Tag.value, res, nullptr, SQConvert::DefSQDestructorCallback<ScriptDateBase>, true);
+		return 1;
+	}
+
+	SQInteger _cmp(HSQUIRRELVM vm)
+	{
+		ScriptDateBase *other = static_cast<ScriptDateBase *>(Squirrel::GetRealInstance(vm, 2, Tag.value));
+		if (other == nullptr) return SQ_ERROR;
+		sq_pushinteger(vm, this->_date - other->_date);
+		return 1;
+	}
+
+protected:
+	Date _date = DATE_INVALID;
```

Comment:
```suggestion
	Date date = DATE_INVALID;
```

Reply:
It conflicts with `Date date()` on line 92.